tests/behat/bootstrap/TideCommonTrait.php

```diff
@@ -168,4 +168,16 @@ public function findMenuItemMatchingText(string $text): void {
     }
   }
 
+  /**
+   * Wait for the Batch API to finish.
+   *
+   * Wait until the id="updateprogress" element is gone,
+   * or timeout after 3 minutes (180,000 ms).
+   *
+   * @Given /^I wait for the batch process to finish$/
+   */
+  public function iWaitForTheBatchProcessToFinish(): void {
+    $this->getSession()->wait(180000, 'jQuery("#updateprogress").length === 0');
```

Comment:
What's so special about 3 minutes? Since it's behat you could pass the value in as part of the Given, so you could say 'I wait 180s for the batch process to finish' and pass it in and multiply by 1000

Reply:
hey @krakerag , nothing special, just give the batch process more times. 
the code mostly copied from https://git.drupalcode.org/project/drupalextension/-/blob/3.2/src/Drupal/DrupalExtension/Context/BatchContext.php?ref_type=heads
but our `FeatureContext` only accepts `Trait`,  so write it into dev-tool

Reply:
Yeah no I get that :)

I'm saying you should make it reusable, so we can pass in the duration as an argument

https://behat.org/en/latest/user_guide/context/definitions.html#defining-reusable-actions

Reply:
sure, looking into it!

Reply:
hey @krakerag , made a change, please take a look, thanks!

Reply:
Love it - thanks!